### DISCUSSION TITLE
How can I use MenuItem as a link?

### DISCUSSION THREAD

**Discussion by IanVS** [ariakit/ariakit#2278] [Questions] [answered]
I'm upgrading to react-router v6, and have run into a problem with `MenuItem`.  I've been using it with an `as={Link}` which worked fine in v5, but now react-router has moved location state into its own prop.  So, I need to pass `state` as a prop to the `Link`, but it's being intercepted by the `MenuItem` itself.

I saw mention of moving to a `forwardProps` or `passProps` in https://github.com/ariakit/ariakit/issues/1245#issuecomment-1359467136, but it doesn't look like that has happened.  There's also no example of using MenuItem as a link, unfortunately.  

Do you have any suggestions?

**Comment by georgekaran** [comment]
Hi there,

One way you could solve this is to spread the props of the `MenuItem` into the `Link`. This way, you can be strict about each state.

Example:
```tsx
<MenuItem
    {...rest}
    state={menuItemState}
>
    {(itemProps: HTMLAttributes<HTMLElement> & RefAttributes<HTMLElement>): ReactElement => (
        <Link
            {...itemProps}
            state={{
                foo: 1,
            }}
        >
            Custom menu item link
        </Link>
    )}
</MenuItem>
```

**Reply by dvnrsn** [reply]
I found this looking for a mechanism for using anchor as menuitem. I'm confused why it's not on the [main documentation page ](https://ariakit.org/components/menu) as it seems to be a very likely use case?

This does fail type check because anchor is more specific than HTMLElement I believe

This works for me but feels wrong
```jsx
  {(
    itemProps: HTMLAttributes<HTMLAnchorElement> &
      RefAttributes<HTMLAnchorElement>,
  ): ReactElement => (
    <Link className="block" {...itemProps} to="/pandora">
      Pandora
    </Link>
  )}

**Reply by diegohaz** [reply, marked as answer]
@dvnrsn You can use the [`render`](https://ariakit.org/guide/composition#changing-the-html-element) prop.

**Reply by dvnrsn** [reply]
Ok that's slick. If I had a good idea of how to hint at it on the linked documentation page I would.

**Reply by diegohaz** [reply]
Yes, we do need to improve that page, maybe with an example similar to [Combobox with links](https://ariakit.org/examples/combobox-links).
